src/Umbraco.Core/Services/ContentTypeServiceBase.cs

```diff
@@ -50,25 +49,30 @@ internal IEnumerable<IContentTypeBase> GetContentTypesForXmlUpdates(params ICont
                     && (dirty.WasPropertyDirty("Alias") || dirty.WasPropertyDirty("HasPropertyTypeBeenRemoved") || hasAnyPropertiesChangedAlias))
                 {
                     //If the alias was changed then we only need to update the xml structures for content of the current content type.
-                    //If a property was deleted or a property alias was changed then we need to update the xml structures for any 
+                    //If a property was deleted or a property alias was changed then we need to update the xml structures for any
                     // content of the current content type and any of the content type's child content types.
                     if (dirty.WasPropertyDirty("Alias")
                         && dirty.WasPropertyDirty("HasPropertyTypeBeenRemoved") == false && hasAnyPropertiesChangedAlias == false)
                     {
-                        //if only the alias changed then only update the current content type                        
+                        //if only the alias changed then only update the current content type
                         toUpdate.Add(contentType);
                     }
                     else
                     {
                         //if a property was deleted or alias changed, then update all content of the current content type
-                        // and all of it's desscendant doc types.     
-                        toUpdate.AddRange(contentType.DescendantsAndSelf(this));
+                        // and all of it's desscendant doc types.
+                        toUpdate.Add(contentType);
+                        toUpdate.AddRange(GetDescendants(contentType));
                     }
                 }
             }
 
             return toUpdate;
+        }
 
+        public virtual IEnumerable<IContentTypeBase> GetDescendants(IContentTypeBase contentType)
+        {
+            return Enumerable.Empty<IContentTypeBase>();
```

Comment:
This method should be overriden in `ContentTypeService` and `MediaTypeService` (IIRC) ie wherever it makes sense to get descendants.